lib/orm.php

```diff
@@ -2541,4 +2580,71 @@ public function __unset( $key ) {
 	public function __isset( $key ) {
 		return $this->offsetExists( $key );
 	}
+
+	/**
+	 * Get a cache.
+	 *
+	 * @param string $key      Cache key.
+	 * @param mixed  $group_id Cache group ID.
+	 *
+	 * @return mixed The cache value.
+	 */
+	public static function get_cache( $key, $group_id = '' ) {
+		$cache_group = self::get_cache_group( $group_id );
+		return \wp_cache_get( $key, $cache_group );
+	}
+
+	/**
+	 * Set a cache.
+	 *
+	 * @param string $key      Cache key.
+	 * @param mixed  $value    Cache value.
+	 * @param mixed  $group_id Cache group ID.
+	 *
+	 * @return void
+	 */
+	private static function set_cache( $key, $value, $group_id = '' ) {
+		/**
+		 * Set the cache, with a timeout of 10 minutes.
+		 *
+		 * The 10-minute is long enough to significantly improve performance,
+		 * but short enough to not cause issues with stale data.
+		 *
+		 * Individual cache-groups are purged when a record is updated or deleted.
+		 */
+		\wp_cache_set( $key, $value, self::get_cache_group( $group_id ), ( 10 * MINUTE_IN_SECONDS ) );
+	}
+
+	/**
+	 * Flushes the cache for the given group IDs.
+	 *
+	 * @param mixed $group_id Cache group ID.
+	 *
+	 * @return void
+	 */
+	public static function flush_group_caches( $group_id = '' ) {
+		// Bail early if the cache doesn't support group-flushing.
+		if ( function_exists( 'wp_cache_supports' ) && ! wp_cache_supports( 'flush_group' ) ) {
+			return;
+		}
+		// Bail early if the `wp_cache_flush_group` function doesn't exist.
+		if ( ! function_exists( 'wp_cache_flush_group' ) ) {
+			return;
+		}
+		\wp_cache_flush_group( self::get_cache_group( $group_id ) );
+	}
+
+	/**
+	 * Get the cache-group to use in cached queries.
+	 *
+	 * @param string $id The ID to use in the cache-group.
+	 *
+	 * @return mixed
+	 */
+	private static function get_cache_group( $id = '' ) {
+		if ( ! $id ) {
+			return 'wp-seo';
+		}
+		return 'wp-seo-' . $id;
```

Comment:
Very minor, but I think we should use the yoast-seo cache group (prefix) instead. Especially since we already use the `yoast-seo-indexable` for the homepage indexable cache.
```suggestion
			return 'yoast-seo';
		}
		return 'yoast-seo-' . $id;
```